public/app/features/scenes/components/VizPanel.tsx

```diff
@@ -19,6 +20,10 @@ export class VizPanel extends SceneObjectBase<VizPanelState> {
   static Component = ScenePanelRenderer;
   static Editor = VizPanelEditor;
 
+  _variableDependency = new VariableDependencyConfig(this, {
+    statePaths: ['title'],
+  });
```

Comment:
When I was thinking about this, I was considering scanning the entire object state looking for variables in those. I guess this would be a lot more versatile (variables could be used anywhere in the state), but would increase the complexity significantly. But the `VariableDependencyConfig` would be just a callback config then.

Reply:
@dprokop 

The reason to be more explicit about what parts of the state we should scan are for an object like SceneQueryRunner which has queries AND data, and the data property can change very frequently and be very big so scanning it could be costly. But maybe that is the only exception. I think the default could probably be to scan the full state, and have an exclude option instead. 

But what I like about this VariableDependencyConfig abstraction is that we could have different implementations, this default one is JSONifying state and looking for variable expressions but others could declare them statically (think of more manually coded data apps that have statically defined variables and dependencies)